usr/sbin/p11sak/p11sak.c

```diff
@@ -1141,6 +1191,200 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
     }
 }
 
+/**
+ *  Print in p11sak_defined_attrs.conf defined attributes in short format
+ */
+static CK_RV short_print_vendor() 
+{
+    CK_RV rc = CKR_OK;
+    int f;
+    FILE *fp;
+    struct ConfigBaseNode *cfg, *c, *name, *hex_string, *type;
+    struct ConfigStructNode *structnode;
+    char file_loc[1024] = P11SAK_DEFINED_ATTRS_LOCATION;
+
+    // get file location
+    cfg = NULL;
+    if (strcmp(P11SAK_DEFINED_ATTRS_LOCATION, "") == 1) {
+        strcpy(file_loc, P11SAK_DEFINED_ATTRS_LOCATION);
+    } else {
+        strcpy(file_loc, P11SAK_DEFAULT_CONF_FILE);
+    }
+
+    // open and parse file
```

Comment:
Are you parsing the config file freshly for every object you print the attributes for? 
With the list operation, you can list hundreds of objects, so I would parse the config file only once, but then operate on the AST for each object to print.

Reply:
Done